src/common/snippets/include/snippets/op/subgraph.hpp

```diff
@@ -187,6 +190,9 @@ class Subgraph : public ov::op::util::SubGraphOp {
         // True if Subgraph contains ops that are not applicable to auto broadcast rule.
         // (e.g. GroupNormalization, reshape)
         bool m_has_broadcast_sensitive_ops = false;
+#ifdef SNIPPETS_DEBUG_CAPS
+        DebugCapsConfig m_debug_config;
```

Comment:
When we store `DebugCapsConfig` by value in two places (op::Subgraph & LIR), there always be a problem of consistency of these two states. For example, what if I call `Subgraph::get_debug_config()` after LIR is created? Can we rely on this value? What if it was already changed in the LIR config?

The easiest way to address this issue is to store `DebugCapsConfig` by pointer in both of the configs. This way we will always automatically maintain consistency. Do you think we can do it?

Reply:
Done